addons/website/views/snippets.xml

```diff
@@ -1224,10 +1224,18 @@
         data-selector="section, .parallax, :not(.o_gallery > .container) > .carousel"
         data-exclude=".s_hr, .s_image_gallery"/>
 
-    <!-- Background Image -->
-    <div data-js="background_position"
-        data-selector="section, .parallax, :not(.o_gallery > .container) > .carousel">
-        <we-button data-background-position="true" data-no-preview="true">Background Image Sizing</we-button>
+    <!-- Background Image Size/Position -->
+    <div data-js="BackgroundPosition"
+        data-selector="section, .carousel-item">
+        <we-select data-no-preview="true">
+            <we-button data-background-type="cover">Cover</we-button>
+            <we-button data-background-type="repeat-pattern">Repeat pattern</we-button>
+        </we-select>
+        <we-input class="o_bg_width_input" string="Background width" data-set-style="background-size" data-default-value="100" data-unit="px"/>
+        <we-input class="o_bg_height_input" string="Background height" data-set-style="background-size" data-default-value="100" data-unit="px"/>
```

Comment:
At the moment my code doesn't handle `auto` values for background-size calculations. It seems kind of weird to me that auto would be supported at all here, auto preserves the background original's aspect ratio, but you can do the same thing with pixel values, and `auto auto` is the image's natural size which you can also just type into the inputs, so the use case is already covered with the numbers.

And then there's the confusion factor of why an input with a unit behind it supports a keyword (`auto px` doesn't mean anything in css) and why it doesn't support other keywords (like contain or initial) or typing in units. IMO these fields should either be unitless and support everything (which as we have discussed is brittle and might break in unexpected ways depending on browsers' standard compliance so it's a bad idea) or, and this is what I think we should do here, have a fixed unit (determined by the xml view as is already the case) and numeric inputs, not supporting any keyword at all.

This has the advantage that, being a native field, it is accessible by default, supports a min and max, has keyboard controls built-in (for which you have a custom handler right now) and supports steps, while only allowing numeric inputs.

Reply:
At the moment, my background size calculation doesn't support `auto` for background-size calculations. Since `auto` preserves the background's aspect ratio, the use case is covered by just inputting the corresponding pixel value, and `auto auto` keeps the image's natural dimensions so you can also just use those pixel values to cover that use case.

And then there's the weirdness that comes from the fact that auto would be supported at all in a field with a unit, `auto px` doesn't mean anything in CSS. What's more, why would it support one keyword in particular but not others (like `contain` or `initial`), or even other units?

As discussed previously, supporting all units is not feasible, and I think that supporting keywords is a bad idea for the same reasons: lots of edge cases and pitfalls. Also, not supporting keywords would allow us to use `<input type="number"/>` instead, which has a good number of advantages, such as input validation, native keyboard controls (instead of needing a custom handler as you have now), support for a step, min and max.

In the cases where we do want to support auto, I think we should use a separate option (like we do here with cover)

Reply:
@sdegueldre 

Clearly, auto should be supported IMO and the system I implemented supported it. In fact the inputs in your task probably do not work because the default value has to be written with the unit. So "100px"... but in this case, it should really be `auto` IMO. If I upload a 48x33 pattern and choose 67px width, I clearly don't expect to have to put 67/(48/33) = 46.0625px in the second input to not have my image deformed (and most people (like 87%) don't want their image deformed).

> And then there's the weirdness that comes from the fact that auto would be supported at all in a field with a unit, auto px doesn't mean anything in CSS. What's more, why would it support one keyword in particular but not others (like contain or initial), or even other units?

-> We don't support the user writing "auto" just the user removing the value so that it default to "auto" and I think this is clear he can't enter words precisely because 'px' is written behind. I think it is clear for most users (like 92%) that when you don't enter anything, it is computed automatically.

> Also, not supporting keywords would allow us to use <input type="number"/> instead

(So as I explained we don't support keywords) but using number inputs is debatable. I personally really don't like the UI being bloated with up and down buttons for each input, and there is the fact it renders differently on all browsers which we try to avoid in Odoo. But overall: number inputs are not supported by Safari on iOS so those users won't have the up/down/min/max/step features I planned to implement.

@sbuhl What is your opinion about all of this?

Reply:
Alright, in that case it's fine to only support auto for height I guess?

Putting the unit in the default value solved the input problem.

Reply:
There is a problem with setting `data-default-value="auto"` for both of them ?

Reply:
Not really, I just thought that since most of the default images are quite big, they would cover the entire background and not make it obvious it's repeating, so for new users it might be a little confusing. But I guess it's most convenient for people who are actually trying to use the feature to just use the image's natural dimensions.

Added some code for that and put both defaults to auto.